lib/Dialect/TritonGPU/Transforms/AccelerateAMDMatmul.cpp

```diff
@@ -284,12 +358,16 @@ class BlockedToMFMA : public mlir::RewritePattern {
         ttg::DotOperandEncodingAttr::get(ctx, 1, mfmaEnc, kWidth));
     a = rewriter.create<ttg::ConvertLayoutOp>(a.getLoc(), newAType, a);
     b = rewriter.create<ttg::ConvertLayoutOp>(b.getLoc(), newBType, b);
-    auto newDot = rewriter.create<tt::DotOp>(dotOp.getLoc(), newRetType, a, b,
-                                             newAcc, dotOp.getAllowTF32(),
-					     dotOp.getMaxNumImpreciseAcc());
+    auto newDot = rewriter.create<tt::DotOp>(dotOp.getLoc(), newAcc.getType(),
+                                             a, b, newAcc, dotOp.getAllowTF32(),
+                                             dotOp.getMaxNumImpreciseAcc());
+
+    Value dotOutput =
+        convertAndCastTensor(rewriter, newDot, oldRetType.getEncoding(),
+                             oldRetType.getElementType());
+
+    rewriter.replaceOp(op, dotOutput);
 
-    rewriter.replaceOpWithNewOp<ttg::ConvertLayoutOp>(op, oldRetType,
-                                                      newDot.getResult());
```

Comment:
changes in this file are copied from upstream PRs: https://github.com/openai/triton/pull/3002 and https://github.com/openai/triton/pull/3025